app/src/main/java/com/github/gotify/service/WebSocketService.java

```diff
@@ -46,16 +62,17 @@
     private AtomicInteger lastReceivedMessage = new AtomicInteger(NOT_LOADED);
     private MissedMessageUtil missingMessageUtil;
 
+    private Picasso cache;
+    private Map<Integer, String> appIdMap;
```

Comment:
This seems not thread save. Use ConcurrentHashMap to ensure thread safety Also the name can be more descriptive.
```suggestion
    private Map<Integer, String> appIdToAppImage = new ConcurrentHashMap<>();
```